### PR TITLE
doc contribution/documentation: Convert the format reST to markdown

### PR DESCRIPTION
**Pull request by otegami**
ref: https://github.com/mroonga/mroonga/issues/707

Since many people are more familiar with Markdown format than reStructuredText format,
We decided to convert the documentation to Markdown to make it easier to maintain.

In this PR, I just convert the file format and fix some links in markdown format.
At the following PRs, We will update the contents each section.

### CODE REVIEW

**Review by abetomo — APPROVED** [review]

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 1]
```diff
@@ -73,13 +73,13 @@ msgstr ""
 msgid "Initial configuration for working repository"
 msgstr ""
 
-msgid "Clone Mroonga repository to working directory. Don't forget to do \"Git configuration\".::"
+msgid "Clone Mroonga repository to working directory. Don't forget to do \"Git configuration\".:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:76]:
```suggestion
msgid "Clone Mroonga repository to working directory. Don't forget to do \"Git configuration\"."
```

`:`はいらないのでは？

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 2]
```diff
@@ -73,13 +73,13 @@ msgstr ""
...
 msgstr ""
 
 msgid "Initial configuration for building documentation"
 msgstr ""
 
-msgid "Execute following commands to prepare for generating Mroonga documentation::"
+msgid "Execute following commands to prepare for generating Mroonga documentation:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:82]:
```suggestion
msgid "Execute following commands to prepare for generating Mroonga documentation."
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 3]
```diff
@@ -91,13 +91,13 @@ msgstr ""
 msgid "Follow the upstream"
 msgstr ""
 
-msgid "Mroonga本家の最新状態に追従して、作業がかぶらないようにします。::"
+msgid "Mroonga本家の最新状態に追従して、作業がかぶらないようにします。:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:94]:
```suggestion
msgid "Mroonga本家の最新状態に追従して、作業がかぶらないようにします。"
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 4]
```diff
@@ -118,31 +115,31 @@ msgstr ""
 msgid "Confirm generated document"
 msgstr ""
 
-msgid "マークアップに問題がないか、HTMLを確認します。HTMLを生成するには以下のコマンドを実行します。::"
+msgid "マークアップに問題がないか、HTMLを確認します。HTMLを生成するには以下のコマンドを実行します。:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:118]:
```suggestion
msgid "マークアップに問題がないか、HTMLを確認します。HTMLを生成するには以下のコマンドを実行します。"
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 5]
```diff
@@ -118,31 +115,31 @@ msgstr ""
...
 msgstr ""
 
-msgid "いつも使っているブラウザで該当ファイルを確認して、変更した内容が反映されていればOKです。::"
+msgid "いつも使っているブラウザで該当ファイルを確認して、変更した内容が反映されていればOKです。:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:121]:
```suggestion
msgid "いつも使っているブラウザで該当ファイルを確認して、変更した内容が反映されていればOKです。"
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 6]
```diff
@@ -118,31 +115,31 @@ msgstr ""
...
 msgstr ""
 
 msgid "Commit"
 msgstr ""
 
-msgid "HTMLに問題がないことを確認できたら、コミットします。::"
+msgid "HTMLに問題がないことを確認できたら、コミットします。:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:127]:
```suggestion
msgid "HTMLに問題がないことを確認できたら、コミットします。"
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 7]
```diff
@@ -118,31 +115,31 @@ msgstr ""
...
 msgstr ""
 
-msgid "コミットするときのメッセージについては、例えば以下のようにします。::"
+msgid "コミットするときのメッセージについては、例えば以下のようにします。:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:130]:
```suggestion
msgid "コミットするときのメッセージについては、例えば以下のようにします。"
```

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po [hunk 8]
```diff
@@ -118,31 +115,31 @@ msgstr ""
...
 msgstr ""
 
 msgid "Push and pull request"
 msgstr ""
 
-msgid "Publish your changes to your own GitHub repository::"
+msgid "Publish your changes to your own GitHub repository:"
```
Comment (komainu8) [inline doc/locale/ja/LC_MESSAGES/contribution/documentation.po:136]:
```suggestion
msgid "Publish your changes to your own GitHub repository"
```

File: doc/source/contribution/documentation.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+
+以下では、最初に一度だけ実施しておけば良いことを説明します。
+
+### Required softwares
+
+TODO
+
+### Git configuration
+
+まずは、gitの設定をしましょう。すでにある程度gitを使っている場合には初期設定はすでに完了しているかも知れません。その場合には飛ばして構いません。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:25]:
```suggestion
まずは、gitの設定をしましょう。すでにある程度gitを使っている場合には初期設定はすでに完了しているかも知れません。その場合には飛ばして構いません。
```

File: doc/source/contribution/documentation.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+```console
+% git config --global user.name "Your Name"
+% git config --global user.email "Email address"
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:29]:
```suggestion
% git config --global user.email "Your email address"
```

File: doc/source/contribution/documentation.md [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+上記はコミットログに使われます。公開しても差し支えないユーザ名もしくはメールアドレスを設定します。
+
+### Fork on GitHub
+
+First, create GitHub account. If your GitHub account is ready, login to GitHub and access following URL.
+
+- [Fork the Mroonga repository](https://github.com/mroonga/mroonga/fork)
+
+Fork リポジトリ選択画面でご自分のリポジトリへとforkしてください。
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:40]:
```suggestion
Fork リポジトリ選択画面で自分のリポジトリへforkしてください。
```

File: doc/source/contribution/documentation.md [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+
+### Initial configuration for working repository
+
+Clone Mroonga repository to working directory. Don't forget to do "Git configuration".:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:44]:
```suggestion
Clone Mroonga repository to working directory. Don't forget to do "Git configuration".
```

File: doc/source/contribution/documentation.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+
+```console
+% git clone git@github.com:(YOUR_GITHUB_ACCOUNT)/mroonga.git
+% cd mroonga
+% git remote add upstream git@github.com:mroonga/mroonga
+```
+
+### Initial configuration for building documentation
+
+Execute following commands to prepare for generating Mroonga documentation:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:54]:
```suggestion
Execute following commands to prepare for generating Mroonga documentation
```

File: doc/source/contribution/documentation.md [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+
+Next step is "The things you need to do every tasks".
+
+## The things you need to do every tasks
+
+以下では作業ごとにやることを説明します。
+
+### Follow the upstream
+
+Mroonga本家の最新状態に追従して、作業がかぶらないようにします。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:69]:
```suggestion
Mroonga本家の最新状態に追従して、作業がかぶらないようにします。
```

File: doc/source/contribution/documentation.md [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+% git checkout -b use-capitalized-notation-characteristic
+```
+
+### Editing text
+
+Fix typos, styles or write a new document for Mroonga.
+
+### Confirm generated document
+
+マークアップに問題がないか、HTMLを確認します。HTMLを生成するには以下のコマンドを実行します。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md:97]:
```suggestion
マークアップに問題がないか、HTMLを確認します。HTMLを生成するには以下のコマンドを実行します。
```

File: doc/source/contribution/documentation.md [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+
+```console
+% cd doc/locale/en
+% make html
+```
+
+いつも使っているブラウザで該当ファイルを確認して、変更した内容が反映されていればOKです。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md]:
```suggestion
いつも使っているブラウザで該当ファイルを確認して、変更した内容が反映されていればOKです。
```

File: doc/source/contribution/documentation.md [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+
+```console
+% firefox html/characteristic.html
+```
+
+### Commit
+
+HTMLに問題がないことを確認できたら、コミットします。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md]:
```suggestion
HTMLに問題がないことを確認できたら、コミットします。
```

File: doc/source/contribution/documentation.md [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+
+```console
+% cd ${cloneしたディレクトリーのトップディレクトリー}
+% git add doc/source/characteristic.rst
+% git commit
+```
+
+コミットするときのメッセージについては、例えば以下のようにします。:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md]:
```suggestion
コミットするときのメッセージについては、例えば以下のようにします。
```

File: doc/source/contribution/documentation.md [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+
+```
+doc: use "Mroonga" notation
+```
+
+### Push and pull request
+
+Publish your changes to your own GitHub repository:
```
Comment (komainu8) [inline doc/source/contribution/documentation.md]:
```suggestion
Publish your changes to your own GitHub repository
```

**Comment by otegami** [comment]
@komainu8 
Thank you for reviewing. I remove unnecessary `:` at this commit: [2ab848f](https://github.com/mroonga/mroonga/pull/717/commits/2ab848f876e91a77604a87378aa885aa141b6599).
About the content, I will fix it in the following PR because I want to focus on converting format in this PR.
Is that okay for you?

**Comment by komainu8** [comment]
> @komainu8 Thank you for reviewing. I remove unnecessary `:` at this commit: [2ab848f](https://github.com/mroonga/mroonga/pull/717/commits/2ab848f876e91a77604a87378aa885aa141b6599). About the content, I will fix it in the following PR because I want to focus on converting format in this PR. Is that okay for you?

Yes, it is.

File: doc/locale/ja/LC_MESSAGES/contribution/documentation.po (hunk 1 above)
Comment (otegami) [reply doc/locale/ja/LC_MESSAGES/contribution/documentation.po]: fix: [2ab848f](https://github.com/mroonga/mroonga/pull/717/commits/2ab848f876e91a77604a87378aa885aa141b6599)

**Review by komainu8 — APPROVED** [review]
+1

**Pull request merged**